Percentage slider -> Aligned min/max dividers

fti-rchaniya opened the pull request:
**Description**
This PR has following enhancement in the PercentageSlider component:

- In our current component, it was noticed in the demo by @h2suzuki san that the alignment of the min (0%) and max(100%) values are not proper as shown in below attached snapshot,
![image](https://github.com/future-standard/scorer-ui-kit/assets/118800413/c86f4e1a-9586-422c-b8a2-35902dde876e)

- So, I have aligned  min (0%) and max(100%) values as shown in the below attached snapshot,
![image](https://github.com/future-standard/scorer-ui-kit/assets/118800413/c01ec055-2a45-4461-acb4-3fa2b4d58a10)

atomworks reviewed (changes requested):
At the moment this is using a nested terniary which makes it very hard to read. It's better to break this logic in to different conditionals so it's easier to identify the cases it's being used for, perhaps instance (or line) for the normal case and then another for the end cases.

Also, I would like to add a prop to this component to make this alignment of the end numbers option (as there are cases where it will need to be aligned as it is originally). Thank you.

fti-rchaniya commented:
> At the moment this is using a nested terniary which makes it very hard to read. It's better to break this logic in to different conditionals so it's easier to identify the cases it's being used for, perhaps instance (or line) for the normal case and then another for the end cases.
> 
> Also, I would like to add a prop to this component to make this alignment of the end numbers option (as there are cases where it will need to be aligned as it is originally). Thank you.

Thank you for your feedback. I appreciate your suggestion to improve readability by breaking down the nested ternary logic. I have refactored the code to use separate conditionals for better clarity, distinguishing between normal and end cases.

Additionally, I have implemented a new prop for the component to provide an option for aligning the end numbers, as you've suggested. This will allow more flexibility in controlling the alignment based on specific use cases.
Kindly verify changes and let me know if it's match your suggestion
Thank you.

atomworks reviewed (commented):
Once Isabel's feedback has been followed, I'm happy to approve this

fti-rchaniya commented on src/Form/molecules/PercentageSlider.tsx:
```diff
@@ -53,6 +53,7 @@ interface IPercentageSliderProps {
   inputCallback?: (value: number) => void
   updateThumbColor?:  (value: number) => IFeedbackColor
   updateTitle?: (value: number) => string
+  isCenterAlignedEndNum: boolean
```
@isacoder, Feedback changes have been implemented, kindly review once again and let me know if any change is required.

Thanks

isacoder reviewed (changes requested):
Hi!  Thank you for updating the Slider Input Story, it's helping with this review.
I see that the last value in the example is very missed aligned. Please modify the code to cover this case too. 

<img width="361" alt="Screen Shot 2023-12-11 at 13 07 49" src="https://github.com/future-standard/scorer-ui-kit/assets/10409078/ed33fd6a-20ba-4b13-83fd-7a947f376d2e">

fti-rchaniya commented:
> Hi! Thank you for updating the Slider Input Story, it's helping with this review. I see that the last value in the example is very missed aligned. Please modify the code to cover this case too.
> 
> <img alt="Screen Shot 2023-12-11 at 13 07 49" width="361" src="https://private-user-images.githubusercontent.com/10409078/289423182-ed33fd6a-20ba-4b13-83fd-7a947f376d2e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDIyODY3ODEsIm5iZiI6MTcwMjI4NjQ4MSwicGF0aCI6Ii8xMDQwOTA3OC8yODk0MjMxODItZWQzM2ZkNmEtMjBiYS00YjEzLTgzZmQtN2E5NDdmMzc2ZDJlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjExVDA5MjEyMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTcwZTE2NzM5NzRhYjJjNzQyYmI0YjM3MDc4MmQ3NTlmMjJlYzViYWMxMjQ1MzM4ZGQ5N2QwNjg2MDIzNWUxYjEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.6Yah3Xsizp6qmAcUuT441JPHCWtkWjwox8wdA-ajbak">





> Hi! Thank you for updating the Slider Input Story, it's helping with this review. I see that the last value in the example is very missed aligned. Please modify the code to cover this case too.
> 
> <img alt="Screen Shot 2023-12-11 at 13 07 49" width="361" src="https://private-user-images.githubusercontent.com/10409078/289423182-ed33fd6a-20ba-4b13-83fd-7a947f376d2e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDIyOTEzOTgsIm5iZiI6MTcwMjI5MTA5OCwicGF0aCI6Ii8xMDQwOTA3OC8yODk0MjMxODItZWQzM2ZkNmEtMjBiYS00YjEzLTgzZmQtN2E5NDdmMzc2ZDJlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjExVDEwMzgxOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWYzZTQwZTNhZWU0NTRiMmFiZThkNWFhOGJlNjM0ZmM5OGMwYTFmZDEzODM5NmE4ZTY0ZjI3MGMzODJkOGY3MjUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.C1V1QhvPDpOYzKdNXa1sX_2U7fnIZlzYaLsxGEkFLuM">



@isacoder san, I have pushed requested changes, can you please check it once and let me know if any changes needed
Thanks:woman-bowing:

isacoder commented:
> > Hi! Thank you for updating the Slider Input Story, it's helping with this review. I see that the last value in the example is very missed aligned. Please modify the code to cover this case too.
> > <img alt="Screen Shot 2023-12-11 at 13 07 49" width="361" src="https://private-user-images.githubusercontent.com/10409078/289423182-ed33fd6a-20ba-4b13-83fd-7a947f376d2e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDIyODY3ODEsIm5iZiI6MTcwMjI4NjQ4MSwicGF0aCI6Ii8xMDQwOTA3OC8yODk0MjMxODItZWQzM2ZkNmEtMjBiYS00YjEzLTgzZmQtN2E5NDdmMzc2ZDJlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjExVDA5MjEyMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTcwZTE2NzM5NzRhYjJjNzQyYmI0YjM3MDc4MmQ3NTlmMjJlYzViYWMxMjQ1MzM4ZGQ5N2QwNjg2MDIzNWUxYjEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.6Yah3Xsizp6qmAcUuT441JPHCWtkWjwox8wdA-ajbak">
> 
> > Hi! Thank you for updating the Slider Input Story, it's helping with this review. I see that the last value in the example is very missed aligned. Please modify the code to cover this case too.
> > <img alt="Screen Shot 2023-12-11 at 13 07 49" width="361" src="https://private-user-images.githubusercontent.com/10409078/289423182-ed33fd6a-20ba-4b13-83fd-7a947f376d2e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDIyOTEzOTgsIm5iZiI6MTcwMjI5MTA5OCwicGF0aCI6Ii8xMDQwOTA3OC8yODk0MjMxODItZWQzM2ZkNmEtMjBiYS00YjEzLTgzZmQtN2E5NDdmMzc2ZDJlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjExVDEwMzgxOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWYzZTQwZTNhZWU0NTRiMmFiZThkNWFhOGJlNjM0ZmM5OGMwYTFmZDEzODM5NmE4ZTY0ZjI3MGMzODJkOGY3MjUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.C1V1QhvPDpOYzKdNXa1sX_2U7fnIZlzYaLsxGEkFLuM">
> 
> @isacoder san, I have pushed requested changes, can you please check it once and let me know if any changes needed Thanks:woman-bowing:

I did a full review of the component, since the update is not working as expected. 
The Slider component was already capable of having the marks centered. This functionality was not available to UIKIT users because of design concerns. However since now is a requirement I think adding the prop to make it available is fine.

I think is better to use the logic that was already in the component since adding extra logic will increase maintenance of the component. I have made a commit to revert the previous logic.

Also Updated the name of the variable isCenterAlignedEndNum -> allMarksCentered to reflect this change.

@fti-rchaniya if you don't have a problem with this update we can move on with this PR

fti-rchaniya commented:
> > > Hi! Thank you for updating the Slider Input Story, it's helping with this review. I see that the last value in the example is very missed aligned. Please modify the code to cover this case too.
> > > <img alt="Screen Shot 2023-12-11 at 13 07 49" width="361" src="https://private-user-images.githubusercontent.com/10409078/289423182-ed33fd6a-20ba-4b13-83fd-7a947f376d2e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDIyODY3ODEsIm5iZiI6MTcwMjI4NjQ4MSwicGF0aCI6Ii8xMDQwOTA3OC8yODk0MjMxODItZWQzM2ZkNmEtMjBiYS00YjEzLTgzZmQtN2E5NDdmMzc2ZDJlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjExVDA5MjEyMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTcwZTE2NzM5NzRhYjJjNzQyYmI0YjM3MDc4MmQ3NTlmMjJlYzViYWMxMjQ1MzM4ZGQ5N2QwNjg2MDIzNWUxYjEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.6Yah3Xsizp6qmAcUuT441JPHCWtkWjwox8wdA-ajbak">
> > 
> > 
> > > Hi! Thank you for updating the Slider Input Story, it's helping with this review. I see that the last value in the example is very missed aligned. Please modify the code to cover this case too.
> > > <img alt="Screen Shot 2023-12-11 at 13 07 49" width="361" src="https://private-user-images.githubusercontent.com/10409078/289423182-ed33fd6a-20ba-4b13-83fd-7a947f376d2e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDIyOTEzOTgsIm5iZiI6MTcwMjI5MTA5OCwicGF0aCI6Ii8xMDQwOTA3OC8yODk0MjMxODItZWQzM2ZkNmEtMjBiYS00YjEzLTgzZmQtN2E5NDdmMzc2ZDJlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjExVDEwMzgxOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWYzZTQwZTNhZWU0NTRiMmFiZThkNWFhOGJlNjM0ZmM5OGMwYTFmZDEzODM5NmE4ZTY0ZjI3MGMzODJkOGY3MjUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.C1V1QhvPDpOYzKdNXa1sX_2U7fnIZlzYaLsxGEkFLuM">
> > 
> > 
> > @isacoder san, I have pushed requested changes, can you please check it once and let me know if any changes needed Thanks:woman-bowing:
> 
> I did a full review of the component, since the update is not working as expected. The Slider component was already capable of having the marks centered. This functionality was not available to UIKIT users because of design concerns. However since now is a requirement I think adding the prop to make it available is fine.
> 
> I think is better to use the logic that was already in the component since adding extra logic will increase maintenance of the component. I have made a commit to revert the previous logic.
> 
> Also Updated the name of the variable isCenterAlignedEndNum -> allMarksCentered to reflect this change.
> 
> @fti-rchaniya if you don't have a problem with this update we can move on with this PR

@isacoder san,

Thank you for your detailed review and efforts on addressing the update issue with the Slider component.

I agree with your decision to leverage the existing logic in the component rather than introducing additional logic to avoid increasing maintenance overhead. 
I support the proposed changes, and let's move forward with this pull request.
Thanks

isacoder reviewed (approved):
Updated and approved
